search/search_repository.go

```diff
@@ -365,6 +523,109 @@ func (r *GormSearchRepository) SearchFullText(ctx context.Context, rawSearchStri
 	return result, count, nil
 }
 
+func (r *GormSearchRepository) listItemsFromDB(ctx context.Context, criteria criteria.Expression, start *int, limit *int) ([]workitem.WorkItemStorage, uint64, error) {
+	where, parameters, compileError := workitem.Compile(criteria)
+	if compileError != nil {
+		return nil, 0, errors.NewBadParameterError("expression", criteria)
+	}
+
+	db := r.db.Model(&workitem.WorkItemStorage{}).Where(where, parameters...)
+	orgDB := db
+	if start != nil {
+		if *start < 0 {
+			return nil, 0, errors.NewBadParameterError("start", *start)
+		}
+		db = db.Offset(*start)
+	}
+	if limit != nil {
+		if *limit <= 0 {
+			return nil, 0, errors.NewBadParameterError("limit", *limit)
+		}
+		db = db.Limit(*limit)
+	}
+
+	db = db.Select("count(*) over () as cnt2 , *").Order("execution_order desc")
+
+	rows, err := db.Rows()
+	if err != nil {
+		return nil, 0, errs.WithStack(err)
+	}
+	defer rows.Close()
+
+	result := []workitem.WorkItemStorage{}
+	columns, err := rows.Columns()
+	if err != nil {
+		return nil, 0, errors.NewInternalError(ctx, err)
```

Comment:
Could you please add some log errors for this kind of internal or unknown errors ? That would help us to track the errors